app/models/notification.rb

```diff
@@ -28,20 +29,29 @@ class Notification < ApplicationRecord
 
   def localized_message
     case purpose
-    when "missed_visit_reminder", "experimental_appointment_reminder"
+    when "covid_medication_reminder"
       I18n.t(
         message,
-        facility_name: subject.facility.name,
+        facility_name: patient.assigned_facility.name,
         patient_name: patient.full_name,
-        appointment_date: subject.scheduled_date,
-        locale: subject.facility.locale
+        locale: patient.assigned_facility.locale
       )
-    when "covid_medication_reminder"
+    when "experimental_appointment_reminder"
+      facility = subject&.facility || patient.assigned_facility
```

Comment:
This is the actual bug fix - make sure we have a facility to determine the locale